From 0a9bcc1dad64951526dfc6ec2e03f5c7fdf0094d Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Ga=C3=ABl=20PORTAY?= <gael.portay@collabora.com>
Date: Tue, 12 May 2020 22:25:16 -0400
Subject: [PATCH] Fix "Use the content of docker variables in an array"

This fixes commit 179675fdf83ab8fbf890a3eb4d18b14ea568a97d.

This commit applies the changes to the command docker exec and docker
run.
---
 dosh | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/dosh b/dosh
index 9906d80..bdfcf8d 100755
--- a/dosh
+++ b/dosh
@@ -506,7 +506,14 @@ fi
 # Execute in existing container
 if [[ ${exec:-} ]]
 then
-	run exec "${docker[@]}" exec "${opts[@]}" ${DOSH_DOCKER_EXEC_EXTRA_OPTS:-} "$exec" "$DOSHELL" "$@"
+	# Append extra options if any
+	read -r -a extraopts < <(echo "${DOSH_DOCKER_EXEC_EXTRA_OPTS:-}")
+	if [ ${#extraopts[*]} -gt 0 ]
+	then
+		opts+=("${extraopts[@]:-}")
+	fi
+
+	run exec "${docker[@]}" exec "${opts[@]}" "$exec" "$DOSHELL" "$@"
 # Run in a new container
 else
 	opts+=("--env" "DOSHLVL=$((DOSHLVL+1))")
@@ -515,5 +522,13 @@ else
 	then
 		echo "$tag"
 	fi >&2
-	run exec "${docker[@]}" run "${opts[@]}" ${DOSH_DOCKER_RUN_EXTRA_OPTS:-} "$tag" "$@"
+
+	# Append extra options if any
+	read -r -a extraopts < <(echo "${DOSH_DOCKER_RUN_EXTRA_OPTS:-}")
+	if [ ${#extraopts[*]} -gt 0 ]
+	then
+		opts+=("${extraopts[@]:-}")
+	fi
+
+	run exec "${docker[@]}" run "${opts[@]}" "$tag" "$@"
 fi
-- 
2.26.2

